### PR TITLE
airbnb lint should not warn on vuex state mutation

### LINKED ISSUE #3954: AirBnb ESLint config, `no-param-reassign` warning in `vuex` mutation
### Version
3.7.0

### Reproduction link
[https://github.com/MaximBazarov/vuex-eslint-mutation-warning](https://github.com/MaximBazarov/vuex-eslint-mutation-warning)

### Environment info
```

  System:
    OS: macOS Mojave 10.14.4
    CPU: (8) x64 Intel(R) Core(TM) i7-3615QM CPU @ 2.30GHz
  Binaries:
    Node: 10.15.3 - /usr/local/bin/node
    Yarn: 1.15.2 - /usr/local/bin/yarn
    npm: 6.4.1 - /usr/local/bin/npm
  Browsers:
    Chrome: 74.0.3729.131
    Firefox: Not Found
    Safari: 12.1
  npmPackages:
    @kazupon/vue-i18n-loader: ^0.3.0 => 0.3.0 
    @vue/babel-helper-vue-jsx-merge-props:  1.0.0-beta.3 
    @vue/babel-plugin-transform-vue-jsx:  1.0.0-beta.3 
    @vue/babel-preset-app:  3.7.0 
    @vue/babel-preset-jsx:  1.0.0-beta.3 
    @vue/babel-sugar-functional-vue:  1.0.0-beta.3 
    @vue/babel-sugar-inject-h:  1.0.0-beta.3 
    @vue/babel-sugar-v-model:  1.0.0-beta.3 
    @vue/babel-sugar-v-on:  1.0.0-beta.3 
    @vue/cli-overlay:  3.7.0 
    @vue/cli-plugin-babel: ^3.7.0 => 3.7.0 
    @vue/cli-plugin-eslint: ^3.7.0 => 3.7.0 
    @vue/cli-plugin-unit-jest: ^3.7.0 => 3.7.0 
    @vue/cli-service: ^3.7.0 => 3.7.0 
    @vue/cli-shared-utils:  3.7.0 
    @vue/component-compiler-utils:  2.6.0 
    @vue/eslint-config-airbnb: ^4.0.0 => 4.0.0 
    @vue/preload-webpack-plugin:  1.1.0 
    @vue/test-utils: 1.0.0-beta.29 => 1.0.0-beta.29 
    @vue/web-component-wrapper:  1.2.0 
    eslint-plugin-vue: ^5.0.0 => 5.2.2 
    jest-serializer-vue:  2.0.2 
    vue: ^2.6.10 => 2.6.10 
    vue-cli-plugin-i18n: ^0.6.0 => 0.6.0 
    vue-eslint-parser:  2.0.3 
    vue-hot-reload-api:  2.3.3 
    vue-i18n: ^8.11.2 => 8.11.2 
    vue-i18n-extract:  0.4.14 
    vue-i18n-ts:  0.1.0 
    vue-jest:  3.0.4 
    vue-loader:  15.7.0 
    vue-router: ^3.0.3 => 3.0.6 
    vue-select: ^3.0.2 => 3.0.2 
    vue-style-loader:  4.1.2 
    vue-template-compiler: ^2.5.21 => 2.6.10 
    vue-template-es2015-compiler:  1.9.1 
    vuex: ^3.0.1 => 3.1.0 
  npmGlobalPackages:
    @vue/cli: Not Found


```

### Steps to reproduce
Vuex store mutation:
```js
state = {...}
```

### What is expected?
no ESLint warnings

### What is actually happening?
ESLint `no-param-reassign` warning

---
it was fixed in 
https://github.com/vuejs-templates/webpack/pull/1003/files#diff-8e97c989db32864ed48507016adcb32d

but seems that it wasn't applied.

### PR DESCRIPTION
**Pull request by LinusBorg**
We had something similiar in the old webpack template:

https://github.com/vuejs-templates/webpack/pull/1003/files#diff-8e97c989db32864ed48507016adcb32d

close #3954

### CODE REVIEW

File: packages/@vue/cli/__tests__/invoke.spec.js [hunk 1]
```diff
@@ -43,7 +43,25 @@ async function assertUpdates (project) {
 
   const eslintrc = parseJS(await project.read('.eslintrc.js'))
   expect(eslintrc).toEqual(Object.assign({}, baseESLintConfig, {
-    extends: ['plugin:vue/essential', '@vue/airbnb']
+    extends: ['plugin:vue/essential', '@vue/airbnb'],
+    rules: {
```
Comment (sodatea) [inline packages/@vue/cli/__tests__/invoke.spec.js]: IMO such duplication in test may not be the best choice. Better use `toMatchObject` and match only an essential subset of the config?

And a corresponding test can be added in the `@vue/cli-plugin-eslint` package to see if a vuex state mutation can pass the eslint check.

Comment (LinusBorg) [reply packages/@vue/cli/__tests__/invoke.spec.js]: ok :)

**Comment by sodatea** [comment]
Just realized this config can be added to `@vue/eslint-config-airbnb` rather than the eslint plugin… And tests are not mandatory for config packages.

**Comment by LinusBorg** [comment]
Understood, will make the changes

**Comment by LinusBorg** [comment]
@sodatea I moved the change to the config-aribnb package as discussed.

Though technically this could be considered a breaking change if we are very picky.

**Comment by sodatea** [comment]
Why?
In regard to ESLint configs, it is only a breaking change when it breaks people's builds. But loosening a rule won't cause more build failures.

**Comment by LinusBorg** [comment]
In that regard I agree. 

If someone is pedantic about what kind of code is committed, they now have a rule pass that might have failed before. 

But I agree that's so minimal, especially in the context of the change, we can discard it.

**Pull request merged**